router/router.go

```diff
@@ -1805,22 +1808,21 @@ func (rt *HandleT) collectMetrics(ctx context.Context) {
 		// failureMetric struct
 		rt.failureMetricLock.RLock()
 		for key, values := range rt.failuresMetric {
```

Comment:
A surge of failed events could cause high memory usage as well (large number of `failureMetric` structs stored in `failuresMetric` map) since we are only collecting metrics on standard time intervals (60s) without caring about their accumulation size.